mne/viz/evoked.py

```diff
@@ -459,6 +432,44 @@ def _plot_lines(data, info, picks, fig, axes, spatial_colors, unit, units,
                 useblit=blit, rectprops=dict(alpha=0.5, facecolor='red'))
 
 
+def _handle_spatial_colors(locs3d, info, idx, chs, type, psd, ax):
+    """Helper for spatial colors."""
+    x, y, z = locs3d.T
+    colors = _rgb(x, y, z)
+    if type in ('meg', 'mag', 'grad', 'eeg'):
+        layout = find_layout(info, ch_type=type, exclude=[])
+    else:
+        layout = find_layout(info, None, exclude=[])
```

Comment:
might as well put `ch_type=None` here to be consistent with the above. And to be DRY you can do
```
ch_type = None if type in (...) else type
layout = find_layout(info, ch_type=ch_type, exclude=[])
```